src/platforms/linux.rs

```diff
@@ -165,45 +128,42 @@ pub fn get_all_apps(search_paths: &[PathBuf]) -> Result<Vec<App>> {
             }
 
             if path.extension().unwrap() == "desktop" && path.is_file() {
-                let (mut app, has_display) = parse_desktop_file(path);
-                // fill icon path if .desktop file contains only icon name
-                if !has_display {
+                let desktop_file_content = std::fs::read_to_string(path)?;
+                let Some((app_name, opt_icon_path)) =
+                    parse_desktop_file_content(&desktop_file_content)
+                else {
                     continue;
-                }
-                if app.icon_path.is_some() {
-                    let icon_path = app.icon_path.clone().unwrap();
-                    if !icon_path.exists() {
-                        // let icon_name = icon_path.file_name().unwrap().to_str().unwrap();
-                        if let Some(icons) = icons_db.get(icon_path.to_str().unwrap()) {
-                            if let Some(icon) = icons.first() {
-                                app.icon_path = Some(icon.path.clone());
-                            }
-                        } else {
-                            app.icon_path = None;
-                        }
-                    }
-                }
+                };
+
+                let app = App {
+                    name: app_name,
+                    icon_path: opt_icon_path,
+                    app_path_exe: None,
+                    app_desktop_path: path.to_path_buf(),
+                };
                 apps.insert(app);
             }
         }
     }
     Ok(apps.iter().cloned().collect())
 }
 
+/// Impl based on https://wiki.archlinux.org/title/Desktop_entries
 pub fn find_all_app_icons() -> Result<HashMap<String, Vec<AppIcon>>> {
-    let hicolor_path: PathBuf = PathBuf::from("/usr/share/icons");
-    let search_dirs = vec![hicolor_path];
+    let mut search_dirs = vec!["/usr/share/icons".into(), "/usr/share/pixmaps".into()];
+    // If $XDG_DATA_DIRS is either not set or empty, a value equal to `/usr/local/share/:/usr/share/` should be used.
+    let xdg_data_dirs =
+        std::env::var("XDG_DATA_DIRS").unwrap_or("/usr/local/share/:/usr/share/".into());
+    for xdg_data_dir in xdg_data_dirs.split(':') {
+        let dir = Path::new(xdg_data_dir).join("icons");
+        search_dirs.push(dir);
+    }
```

Comment:
This function is refactored as well, but we don't use it, we use the utility provided by https://github.com/BL-CZY/desktop_file_parser